app.js

```diff
@@ -79,106 +142,108 @@ app.use(
 
-// Routers
+// Routers-Getters
 
-app.get("/", (req, res) => {
-  res.status(200);
-  const { userID } = req.session;
+app.get("/signin", (req, res) => {
 
+  const { userID } = req.session;
   if (!userID) {
-    renderer(res, "signin");
+    renderer(req, res, "signin");
   } else {
-    renderer(res, "home");
+    renderer(req, res, "home");
+  }
+  
+});
+app.get("/register", (req, res) => {
+  const { userID } = req.session;
+  if (!userID) {
+    renderer(req, res, "register");
+  } else {
+    renderer(req, res, "home");
   }
 });
 
-app.get("/home", redirectLogin, (req, res) => {
-  res.status(200);
-  renderer(res, "home");
+app.get("/home", redirectLogin,(req, res) => {
+  message = null;
+  renderer(req, res, "home");
 });
 
 app.get("/products", redirectLogin, (req, res) => {
-  res.status(200);
-  renderer(res, "products");
+  renderer(req, res, "products");
 });
 
 app.get("/about", redirectLogin, (req, res) => {
-  res.status(200);
-  renderer(res, "about");
-});
-
-/*
-app.get("/register", (req, res) => {
-  res.status(200);
-  renderer(res, "register");
+  renderer(req, res, "about");
 });
 
-app.get("/signin", redirectHome, (req, res) => {
-  res.status(200);
-
-  renderer(res, "signin");
-});
-*/
-
-app.post("/register", redirectHome, (req, res) => {
-  const { name, email, password } = req.body;
-
-  const exist = users.some((el) => {
-    el.email === email;
-  });
-
-  if (!exist) {
-    const user = {
-      id: users.length + 1,
-      name,
-      password,
-      email,
-    };
-    users.push(user);
-
-    req.session.userID = user.id;
-    return res.redirect("/home");
+app.get("/product/:id", redirectLogin, (req, res) => {
+  PID = req.params.id * 1;
+  const product = productsData.products.find((el) => el.id === PID);
+  if (product) {
+    return renderer(req, res, "product");
   }
-
-  res.redirect("/");
+  message = "Product not found in the database !";
+  res.redirect("/404");
 });
 
-app.post("/signin", redirectHome, (req, res) => {
-  const { email, password } = req.body;
+// Routers-Posters
 
-  console.log(req.body);
-  console.log(email, password);
-  console.log(users[0].email, users[0].password);
+app.post("/signin", redirectHome,async (req, res) => {
+  const { email, password } = req.body;
 
   if (email && password) {
-    const user = users.find(
-      (el) => el.email === email && el.password === password
-    );
+    const user = await findUserByEmail(email);
     if (user) {
-      req.session.userID = user.id;
-      console.log("userID =" + req.session.userID);
+      req.session.userID = user._id;
       return res.redirect("/home");
     }
   }
+  message = "Wrong Email/Password !";
+  res.redirect("/signin");
+});
 
-  res.redirect("/");
+app.post("/register", redirectHome,async (req, res) => {
+  const { name, email, password, confirmPassword } = req.body;
+  if (password === confirmPassword) {
+    const extst = await findUserByEmail(email);
```

Comment:
better to use on level of nesting if possible. 
`if (password !== confirmPassword) return res.status()...`